src/publish/s3.js

```diff
@@ -3,9 +3,22 @@ const winston = require("winston-color")
 const zlib = require("zlib")
 const config = require("../config")
 
-const S3 = new AWS.S3()
+let conf = {}
 
+// This is the case where using custom s3 api-like services like minio.
+if (config.aws.endpoint) {
+  conf = {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    endpoint: (config.aws.endpoint ? config.aws.endpoint : ''),
+    s3ForcePathStyle: 'true',
+    signatureVersion: 'v4'
+  }
```

Comment:
Anyway to me it's safer keeping the 2 configurations separated, one for the standard one and the block for servers like minio which need set of custom parameters like `s3ForcePathStyle` or `signatureVersion` which cannot be the standards of S3.
For the ternary operator, reading the code again it was useless because we already are in a if block.